---
title: "HCM_12k_Associations"
output: md_document
always_allow_html: true
code_folding: "hide"
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, error =TRUE)
```
# 3D Associations

The visualisation of cardiac imaging phenotypes derived from UK Biobank cardiovascular magnetic resonance (CMR) images across 50,000 subjects and to enable the exploration of their links to other non-imaging phenotypes that characterise the structure and function for the the left ventricle (LV).\
Wall thickness (WT) and surface to surface (S2S) were calculated at over 40,000 points in the 3D model at end-diastole and were measured as the distance between the endocardial and epicardial surfaces perpendicular to the midwall plane and as the differences in the endocardial and epicardial surfaces relative to an average cardiac shape, respectively.

The associations between one imaging phenotype (WT, S2S) and one non-imaging phenotype for each point in the 3D datasets were assessed using a linear regression model adjusted for Age, Sex, Race, BSA, SBP, DBP.\
Multiple testing with Benjamini-Hochberg (BH) procedure to control from false discovery rate (FDR) and the threshold-free cluster enhancement (TFCE) technique to boost the areas of signal with spatial contiguity, was used in the LV points with significant associations ([Biffi et al., 2018](https://academic.oup.com/bioinformatics/article/34/1/97/4103396)).

From the table summarytable_meta.txt,\
Step 1: 2709 people any rare PAV (syndromic or not)\
Step 2A: 774 people have a rare PAV in the 8 sarcomere genes\
Step 2B: 424 people have a rare PAV (FAF for HCM) variant in the 8 sarcomere genes\
Step 3A: 24 people have a P/LP (CC) rare (FAF) variant in the 8 sarcomere genes\
Step 3B: 18 people have a ClinVar HCM rare (FAF) P/LP variant in the 8 sarcomere genes\
Step 3A_3B: 30 people have a P/LP (CC) rare (FAF) variant a ClinVar HCM rare (FAF) P/LP variant in the 8 sarcomere genes\
Step 4: 12 people have a P/LP (CC), ClinVar, rare (FAF) variant in the 8 sarc genes\
were taken against \
Step 0 and hasanyCM: 9204 people have no rare PAV = 1 and has any CM flag by Kathryn’s analysis |FALSE,\
and their associations with imaging phenotypes in the 3D LV model, were explored from a subset of 12k subjects.\

PAV = protein altering variant\
FAF = filtering allele frequency\
P/LP = flagged as pathogenic or likely pathogenic\
CC = CardioClassifier\

```{r, message=FALSE, warning=FALSE}
summarytable_meta <- read.table("summarytable_meta.txt", header = TRUE)
head(summarytable_meta)
```

## Step 1:

2,523 people any rare PAV (syndromic or not) vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}


library(plotly)
Beta_wt<-as.data.frame(read.table("step1/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta1")
Pvalues_wt<-as.data.frame(read.table("step1/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues1")
Beta_shape<-as.data.frame(read.table("step1/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta1")
Pvalues_shape<-as.data.frame(read.table("step1/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues1")

position<-which(Pvalues_wt$pvalues1< 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene',
            marker = list(color = ~beta1, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.06,
                          cmax = 0.06, showscale = F)) %>%
layout(showlegend = F)
position2<-which(Pvalues_shape$pvalues1 < 0.05)
sig2 <- (length(position2)/nrow(Beta_shape))*100
significance_area2<-format(round(sig2, 0), nsmall = 0)
average_beta2 <- format(round(mean(Beta_shape[position2,4]), 3), nsmall = 2)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene2',
            marker = list(color = ~beta1, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.06,
                          cmax = 0.06, showscale = T)) %>%
layout(showlegend = F)
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)

            
```

## Step 2A:

713 people have a rare PAV in the 8 sarcomere genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step2a/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta2a")
Pvalues_wt<-as.data.frame(read.table("step2a/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues2a")
Beta_shape<-as.data.frame(read.table("step2a/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta2a")
Pvalues_shape<-as.data.frame(read.table("step2a/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues2a")

position<-which(Pvalues_wt$pvalues2a< 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene',
            marker = list(color = ~beta2a, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.1,
                          cmax = 0.1, showscale = F)) %>%
layout(showlegend = F)
position2<-which(Pvalues_shape$pvalues2a < 0.05)
sig2 <- (length(position2)/nrow(Beta_shape))*100
significance_area2<-format(round(sig2, 0), nsmall = 0)
average_beta2 <- format(round(mean(Beta_shape[position2,4]), 3), nsmall = 2)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene2',
            marker = list(color = ~beta2a, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.1,
                          cmax = 0.1, showscale = T)) %>%
layout(showlegend = F)
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```

## Step 2B:

395 people have a rare PAV (FAF for HCM) variant in the 8 sarcomere genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step2b/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta2b")
Pvalues_wt<-as.data.frame(read.table("step2b/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues2b")
Beta_shape<-as.data.frame(read.table("step2b/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta2b")
Pvalues_shape<-as.data.frame(read.table("step2b/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues2b")

position<-which(Pvalues_wt$pvalues2b < 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z,height=600, width = 1200,scene='scene',
            marker = list(color = ~beta2b,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.15,
                          cmax = 0.15, showscale = F)) %>%
  add_markers() 
position2<-which(Pvalues_shape$pvalues2b < 0.05)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600,width = 1200,scene='scene2',
            marker = list(color = ~beta2b,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.15,
                          cmax = 0.15, showscale = T)) %>%
  add_markers() 
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```
```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step2ab/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta2ab")
Pvalues_wt<-as.data.frame(read.table("step2ab/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues2ab")
Beta_shape<-as.data.frame(read.table("step2ab/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta2ab")
Pvalues_shape<-as.data.frame(read.table("step2ab/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues2ab")

position<-which(Pvalues_wt$pvalues2ab < 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z,height=600, width = 1200,scene='scene',
            marker = list(color = ~beta2ab,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.1,
                          cmax = 0.1, showscale = F)) %>%
  add_markers() 
position2<-which(Pvalues_shape$pvalues2ab < 0.05)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600,width = 1200,scene='scene2',
            marker = list(color = ~beta2ab,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.1,
                          cmax = 0.1, showscale = T)) %>%
  add_markers() 
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)

            
```

## Step 3A:

23 people have a P/LP (CC) rare (FAF) variant in the 8 sarcomere genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step3a/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta3a")
Pvalues_wt<-as.data.frame(read.table("step3a/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues3a")
Beta_shape<-as.data.frame(read.table("step3a/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta3a")
Pvalues_shape<-as.data.frame(read.table("step3a/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues3a")

position<-which(Pvalues_wt$pvalues3a < 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene',
            marker = list(color = ~beta3a, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -4,
                          cmax = 4, showscale = F)) %>%
  add_markers() %>%
  add_trace(z = Beta_wt[position,3],
            x = Beta_wt[position,1],
            y = Beta_wt[position,2],type = "scatter3d", mode = "markers",inherit = T,
            marker = list(color =  Beta_wt[position,4],colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.5,
                          cmax = 0.5, showscale = T))%>%layout(showlegend = F)
position2<-which(Pvalues_shape$pvalues3a < 0.05)
sig2 <- (length(position2)/nrow(Beta_shape))*100
significance_area2<-format(round(sig2, 0), nsmall = 0)
average_beta2 <- format(round(mean(Beta_shape[position2,4]), 3), nsmall = 2)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z,height=600, width = 1200,scene='scene2',
            marker = list(color = ~beta3a, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -6,
                          cmax = 6, showscale = F)) %>%
  add_markers() %>%
  add_trace(z = Beta_shape[position2,3],
            x = Beta_shape[position2,1],
            y = Beta_shape[position2,2],type = "scatter3d", mode = "markers",inherit = T,
            marker = list(color =  Beta_shape[position2,4],colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.5,
                          cmax = 0.5, showscale = F))%>%layout(showlegend = F)
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with average beta coefficient = ",average_beta, "and significant area = ", significance_area, "%\nand Shape vs HCM (right) with average beta coefficient = ",average_beta2, "and significant area = ", significance_area2, "%."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```

## Step 3B:

17 people have a ClinVar HCM rare (FAF) P/LP variant in the 8 sarcomere genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step3b/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta3b")
Pvalues_wt<-as.data.frame(read.table("step3b/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues3b")
Beta_shape<-as.data.frame(read.table("step3b/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta3b")
Pvalues_shape<-as.data.frame(read.table("step3b/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues3b")

position<-which(Pvalues_wt$pvalues3b< 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene',
            marker = list(color = ~beta3b, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.6,
                          cmax = 0.6, showscale = F)) %>%
layout(showlegend = F)
position2<-which(Pvalues_shape$pvalues3b < 0.05)
sig2 <- (length(position2)/nrow(Beta_shape))*100
significance_area2<-format(round(sig2, 0), nsmall = 0)
average_beta2 <- format(round(mean(Beta_shape[position2,4]), 3), nsmall = 2)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene2',
            marker = list(color = ~beta3b, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.6,
                          cmax = 0.6, showscale = T)) %>%
layout(showlegend = F)
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```


## Step 3A_3B:

28 people have a P/LP (CC) rare (FAF) variant a ClinVar HCM rare (FAF) P/LP variant in the 8 sarcomere genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step3ab/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta3ab")
Pvalues_wt<-as.data.frame(read.table("step3ab/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues3ab")
Beta_shape<-as.data.frame(read.table("step3ab/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta3ab")
Pvalues_shape<-as.data.frame(read.table("step3ab/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues3ab")

position<-which(Pvalues_wt$pvalues3ab < 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z, height=600, width = 1200,scene='scene',
            marker = list(color = ~beta3ab, colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -4,
                          cmax = 4, showscale = F)) %>%
  add_markers() %>%
  add_trace(z = Beta_wt[position,3],
            x = Beta_wt[position,1],
            y = Beta_wt[position,2],type = "scatter3d", mode = "markers",inherit = T,
            marker = list(color =  Beta_wt[position,4],colorbar=list(title='Beta coefficient - (a.u.)',
                                                                     len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.5,
                          cmax = 0.5, showscale = T))%>%layout(showlegend = F)
position2<-which(Pvalues_shape$pvalues3ab < 0.05)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600,width = 1200,scene='scene2',
            marker = list(color = ~beta3ab,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.6,
                          cmax = 0.6, showscale = T)) %>%
  add_markers() 
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with average beta coefficient = ",average_beta, "and significant area = ", significance_area, "%\nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```

## Step 4:

12 people have a P/LP (CC), ClinVar, rare (FAF) variant in the 8 sarc genes vs 8,696 people have no rare PAV and no CM flag by Kathryn’s analysis (clean set of non imaging phenotypes).

```{r, echo=FALSE, message=FALSE, warning = FALSE}

library(plotly)
Beta_wt<-as.data.frame(read.table("step4/HCM_beta_WT.txt"))
colnames(Beta_wt)<- c("x", "y", "z", "beta4")
Pvalues_wt<-as.data.frame(read.table("step4/HCM_BHpvaluesTFCE_WT.txt"))
colnames(Pvalues_wt) <- c("x", "y", "z", "pvalues4")
Beta_shape<-as.data.frame(read.table("step4/HCM_beta_S2S.txt"))
colnames(Beta_shape)<- c("x", "y", "z", "beta4")
Pvalues_shape<-as.data.frame(read.table("step4/HCM_BHpvaluesTFCE_S2S.txt"))
colnames(Pvalues_shape) <- c("x", "y", "z", "pvalues4")

position<-which(Pvalues_wt$pvalues4 < 0.05)
sig <- (length(position)/nrow(Beta_wt))*100
significance_area<-format(round(sig, 0), nsmall = 0)
average_beta <- format(round(mean(Beta_wt[position,4]), 3), nsmall = 2)
ax <- list(title = "",zeroline = F,showline = F,showticklabels = F,showgrid = F)
p1<-plot_ly(Beta_wt, x = ~x , y = ~y , z = ~z,height=600, width = 1200,scene='scene',
            marker = list(color = ~beta4,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.6,
                          cmax = 0.6, showscale = F)) %>%
  add_markers() 
position2<-which(Pvalues_shape$pvalues4 < 0.05)
p2<-plot_ly(Beta_shape, x = ~x , y = ~y , z = ~z, height=600,width = 1200,scene='scene2',
            marker = list(color = ~beta4,colorbar=list(title='Beta coefficient - (a.u.)',
                                                        len = 0.55, x = 1.13, y = 0.6),
                          colorscale = c('#1972A4', '#FF7070'),
                          cauto= F,
                          cmin = -0.7,
                          cmax = 0.7, showscale = T)) %>%
  add_markers() 
  
p<-subplot(p1,p2) %>%
  layout(title = paste("\nWall thickness vs HCM (left) with no significant associations \nand Shape vs HCM (right) with no significant associations."),
         scene = list(domain=list(x=c(0,0.48),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),
         scene2 = list(domain=list(x=c(0.48,1),y=c(0.1,1)),xaxis = ax,yaxis = ax, zaxis = ax, dragmode="turntable"),showlegend = FALSE)


            
```
